daemon/api_sideload_n_try.go

```diff
@@ -75,24 +77,25 @@ func (f *Form) RemoveAllExcept(paths []string) {
 	}
 }
 
-// SnapFileNameAndTempPath returns the original file path/name and the path to
-// where the temp file is written.
-func (f *Form) SnapFileNameAndTempPath() (srcFilename, path string, apiErr *apiError) {
+// GetSnapFiles returns the original name and temp path for each snap file in the form.
+func (f *Form) GetSnapFiles() (srcFilenames, tempPaths []string, apiErr *apiError) {
 	if len(f.FileRefs["snap"]) == 0 {
-		return "", "", BadRequest(`cannot find "snap" file field in provided multipart/form-data payload`)
+		return nil, nil, BadRequest(`cannot find "snap" file field in provided multipart/form-data payload`)
 	}
 
-	snapFile := f.FileRefs["snap"][0]
-	srcFilename, path = snapFile.Filename, snapFile.TmpPath
+	for _, ref := range f.FileRefs["snap"] {
+		srcFilenames = append(srcFilenames, ref.Filename)
+		tempPaths = append(tempPaths, ref.TmpPath)
+	}
 
-	if len(f.Values["snap-path"]) > 0 {
-		srcFilename = f.Values["snap-path"][0]
+	if len(srcFilenames) == 1 && len(f.Values["snap-path"]) > 0 {
+		srcFilenames[0] = f.Values["snap-path"][0]
```

Comment:
I wonder if we should actively complain if this is the case.

Reply:
You mean return 400 if there's more than one snap and a snap-path? Hm not sure, don't really have a strong opinion. The original path was optional and only used in the error/change summary so there's not much impact either way I think